sky/cli.py

```diff
@@ -4225,22 +4309,126 @@ def serve():
+@click.option(
+    '--disk-tier',
+    default=None,
+    type=click.Choice(['low', 'medium', 'high'], case_sensitive=False),
+    required=False,
+    help=(
+        'OS disk tier. Could be one of "low", "medium", "high". Default: medium'
+    ))
+# Service options.
 @click.option('--service-name',
               '-n',
               default=None,
               type=str,
               help='A service name. Unique for each service. If not provided, '
               'a unique name is autogenerated.')
+@click.option('--readiness-path',
+              default=None,
+              type=str,
+              required=False,
+              help='Path for the readiness probe.')
+@click.option('--initial-delay-seconds',
+              default=None,
+              type=int,
+              required=False,
+              help=('Initial delay in seconds. Any readiness probe failures '
+                    'during this period will be ignored.'))
+@click.option('--min-replicas',
+              default=None,
+              type=int,
+              required=False,
+              help='Minimum number of replicas.')
+@click.option('--max-replicas',
+              default=None,
+              type=int,
+              required=False,
+              help=('Maximum number of replicas. If not specified, SkyServe '
+                    'will use fixed number of replicas same as min_replicas '
+                    'and ignore any specified QPS threshold.'))
+@click.option('--target-qps-per-replica',
+              default=None,
+              type=float,
+              required=False,
+              help=('Target number of queries per second per replica for '
+                    'autoscaling. If not specified, SkyServe will use fixed '
+                    'number of replicas same as min_replicas.'))
+@click.option('--post-data',
+              default=None,
+              type=str,
+              required=False,
+              help=('Post data for the readiness probe. If not specified, '
+                    'use GET request; otherwise, use POST request with '
+                    'the argument as the post data. The argument should be '
+                    'an JSON formatted string, like \'{"key": "value"}\'.'))
+@click.option('--upscale-delay-seconds',
+              default=None,
+              type=int,
+              required=False,
+              help=('Upscale delay in seconds.'))
+@click.option('--downscale-delay-seconds',
+              default=None,
+              type=int,
+              required=False,
+              help=('Downscale delay in seconds.'))
 @click.option('--yes',
               '-y',
               is_flag=True,
               default=False,
               required=False,
               help='Skip confirmation prompt.')
-# TODO(tian): Support the task_option overrides for the service.
+@timeline.event
+@usage_lib.entrypoint
 def serve_up(
     service_yaml: List[str],
     service_name: Optional[str],
+    name: Optional[str],
```

Comment:
it's a bit confusing to have both `name` and `service_name`.

Reply:
can we just have one?

Reply:
We've had an offline discussion about this; just DM-ed the discussion record